compiler/rustc_hir_analysis/src/check/intrinsicck.rs

```diff
@@ -145,8 +145,13 @@ impl<'a, 'tcx> InlineAsmCtxt<'a, 'tcx> {
         if !ty.is_copy_modulo_regions(self.tcx, self.param_env) {
             let msg = "arguments for inline assembly must be copyable";
             let mut err = self.tcx.sess.struct_span_err(expr.span, msg);
-            err.note(format!("`{ty}` does not implement the Copy trait"));
-            err.emit();
+
+            if ty.references_error() {
```

Comment:
There's an arm above:

```rust
            ty::Error(_) => return None,
```

I think it would be better if we just did `_ if ty.references_error()` over there?

Reply:
Yeah.